From 83a883626def335e8f2ce08128d8f21e52826ae0 Mon Sep 17 00:00:00 2001
From: "Leonidas P. Papadakos" <papadakospan@gmail.com>
Date: Fri, 1 Mar 2019 19:54:53 +0200
Subject: [PATCH 06/12] arm64: dts: rockchip: add more cpu operating points for
 RK3328

This allows for greater max frequency on rk3328 boards,
increasing performance.

It has been included in Armbian (a linux distibution for ARM boards)
for a while now without any reported issues

https://github.com/armbian/build/blob/master/patch/kernel/rockchip64-default/enable-1392mhz-opp.patch
https://github.com/armbian/build/blob/master/patch/kernel/rockchip64-default/enable-1512mhz-opp.patch

Cc: Heiko Stuebner <heiko@sntech.de>
Cc: linux-rockchip@lists.infradead.org
Signed-off-by: Leonidas P. Papadakos <papadakospan@gmail.com>
---
 arch/arm64/boot/dts/rockchip/rk3328.dtsi | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3328.dtsi b/arch/arm64/boot/dts/rockchip/rk3328.dtsi
index 1a86ee0a3..e4cda9c77 100644
--- a/arch/arm64/boot/dts/rockchip/rk3328.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3328.dtsi
@@ -140,6 +140,20 @@ opp-1296000000 {
 			opp-microvolt = <1300000>;
 			clock-latency-ns = <40000>;
 		};
+		opp-1392000000 {
+			opp-hz = /bits/ 64 <1392000000>;
+			opp-microvolt = <1350000 1350000 1350000>;
+			opp-microvolt-L0 = <1350000 1350000 1350000>;
+			opp-microvolt-L1 = <1325000 1325000 1325000>;
+			clock-latency-ns = <40000>;
+		};
+		opp-1512000000 {
+			opp-hz = /bits/ 64 <1512000000>;
+			opp-microvolt = <1450000 1450000 1450000>;
+			opp-microvolt-L0 = <1450000 1450000 1450000>;
+			opp-microvolt-L1 = <1425000 1425000 1425000>;
+			clock-latency-ns = <40000>;
+		};
 	};
 
 	amba {
-- 
2.26.0

